lms/djangoapps/courseware/block_render.py

```diff
@@ -97,6 +98,13 @@
 from common.djangoapps.xblock_django.user_service import DjangoXBlockUserService
 from openedx.core.lib.cache_utils import CacheService
 
+if TYPE_CHECKING:
+    from rest_framework.request import Request
+    from xblock.core import XBlock
+    from xblock.runtime import Runtime
+
+    from xmodule.course_block import CourseBlock
```

Comment:
Why are these conditional imports?

Reply:
@ormsbee, [typing.TYPE_CHECKING](https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING) (defined in [PEP-484](https://peps.python.org/pep-0484/)) is `False` at runtime. It should only be `True` during static analysis (usually in an IDE or type checker). Thanks to this, and the fact we don't evaluate annotations in function definitions ([PEP-563](https://peps.python.org/pep-0563/)), we:
1. Reduce the number of imports. These imports are not heavy, but we're not instantiating these classes, so we don't need them at runtime.
2. Avoid potential circular imports at runtime.